packages/openlayers/lib/map/register-events.ts

```diff
@@ -152,6 +154,43 @@ export function listen<T extends keyof MapEventsByType>(
         });
       });
       break;
+    case SourceLoadErrorType:
+      //attach event listener to all existing layers
+      map.getLayers().forEach((layer) => {
+        if (layer) {
+          layer.on(
+            SourceLoadErrorType as unknown as BaseLayerObjectEventTypes,
+            (event: BaseEvent) => {
+              (callback as (event: unknown) => void)(event);
+            },
+          );
+        }
+      });
+      //attach event listener when layer is added
+      map.getLayers().on("add", (event) => {
+        const layer = event.element as Layer;
+        if (layer) {
+          layer.on(
+            SourceLoadErrorType as unknown as BaseLayerObjectEventTypes,
+            (event: BaseEvent) => {
+              (callback as (event: unknown) => void)(event);
+            },
+          );
+        }
+      });
+      //remove event listener when layer is removed
+      map.getLayers().on("remove", (event) => {
+        const layer = event.element as Layer;
+        if (layer) {
+          layer.un(
+            SourceLoadErrorType as unknown as BaseLayerObjectEventTypes,
+            (event: BaseEvent) => {
+              (callback as (event: unknown) => void)(event);
+            },
```

Comment:
Sorry I forgot about this: I don't think this will unregister the listener; you have to give it the same callback that was used when registering the listener

Reply:
Ah good catch, does af8efe0ee4ea27e125e02f722503cc27191bc637 make sense?